rts/motoko-rts/src/gc/incremental/partitioned_heap.rs

```diff
@@ -56,16 +52,20 @@ use super::{
 
 /// Size of each partition.
 /// Select the size of the power of two with the smallest WASM memory size in the benchmark.
-/// -> Small partitions below 32 MB are inefficient in terms of both memory and runtime costs
+/// -> Small partitions below 64 MB are inefficient in terms of both memory and runtime costs
 ///    due to the increased frequency of large object handling.
-/// -> Large partitions above 32 MB are a waste for small programs, since the WASM memory is
+/// -> Large partitions above 64 MB are a waste for small programs, since the WASM memory is
 ///    allocated in that granularity and GC is then triggered later.
-pub const PARTITION_SIZE: usize = 32 * MB;
+pub const PARTITION_SIZE: usize = 64 * 1024 * 1024;
+
+// TODO: Redesign for 64-bit support by using a dynamic partition list.
+/// Currently limited to 64 GB.
+pub const MAXIMUM_MEMORY_SIZE: Bytes<usize> = Bytes(64 * 1024 * 1024 * 1024);
 
 /// Total number of partitions in the memory.
 /// For simplicity, the last partition is left unused, to avoid a numeric overflow when
```

Comment:
Can that overflow still happen for 64-bit? Not that it matters much.

Reply:
I would need to redesign the partition table of the GC (have it on my backlog). Currently, it uses a static table and then has a fixed size. 

The limit should be properly checked in the allocator, avoiding any numeric overflows. I also added a debug assertion in the allocator that the GC memory reserve is less (or equal) the maximum memory size.

Reply:
Maybe, it would be wise to redesign the partition table to a dynamic layout, in combination with the EOP release. Otherwise, it would require a graph copy migration, if we do it later. However, the PRs are already large, and maybe it is better to postpone that complexity, even if it means to migrate by graph copy. I am not sure...

Reply:
We can also do it after the merge, but before the 64-bit release.

Reply:
**Update**: I prepared https://github.com/dfinity/motoko/pull/4556 to overcome the limit.